libsplinter/src/admin/service/shared.rs

```diff
@@ -400,8 +418,12 @@ impl AdminServiceShared {
                     }
                     Ok(CircuitProposalStatus::Rejected) => {
                         // remove circuit
-                        self.remove_proposal(&circuit_id)?;
-
+                        let proposal = self.remove_proposal(&circuit_id)?;
+                        if let Some(proposal) = proposal {
+                            for member in proposal.get_circuit_proposal().members.iter() {
+                                self.remove_peer_ref(member.get_node_id());
```

Comment:
Is there a peer ref per proposal?  Looking at the add_peer_ref, there's a unique peer ref per peer in this state.  Shouldn't we only remove the peer ref if no other proposal has that peer?

Reply:
add_peer_ref adds the peer_ref to a list of peer_refs for that peer_id. remove_peer_ref pops one peer ref off, only keeping the list around if its not empty. This does not remove the peer simply drops one of the peer refs (which will remove a 1 reference from the peer manager)